manifests/manifests.go

```diff
@@ -203,6 +203,18 @@ func (l *list) Push(ctx context.Context, dest types.ImageReference, options Push
 		}
 	}()
 
+	// If we were given a media type that corresponds to a multiple-images
+	// type, reset it to a valid corresponding single-image type, since we
+	// already expect the image library to infer the list type from the
+	// image type that we're telling it to force.
+	singleImageManifestType := options.ManifestType
+	switch singleImageManifestType {
+	case v1.MediaTypeImageIndex:
```

Comment:
I'll just note that I'd prefer 'v1' to be `imgspecv1` in this file as it is in cmd/buildah/manifest.go.